From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 27 Sep 2010 20:35:17 -0400
Subject: [net] qlcnic: TSO feature added for vlan devices
Message-id: <20100927203516.3859.71521.sendpatchset@localhost.localdomain>
Patchwork-id: 28434
O-Subject: [RHEL 5.6 PATCH 5/6] qlcnic: TSO feature added for vlan devices.
Bugzilla: 625084
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla
--------

Bug 625084 (https://bugzilla.redhat.com/show_bug.cgi?id=625084)

Upstream Status
---------------

RHEL 5 specific.

Description
-----------

>From fdd8bd0c2d0929ee9c3201cec2d7dc3c9a35a133 Mon Sep 17 00:00:00 2001
From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Date: Wed, 18 Aug 2010 00:22:43 -0700
Subject: [PATCH] qlcnic: TSO feature added for vlan devices.

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: Amit Kumar Salecha <amit.salecha@qlogic.com>

diff --git a/drivers/net/qlcnic/qlcnic_main.c b/drivers/net/qlcnic/qlcnic_main.c
index f6c5d8c..d4ac0d2 100644
--- a/drivers/net/qlcnic/qlcnic_main.c
+++ b/drivers/net/qlcnic/qlcnic_main.c
@@ -1279,10 +1279,11 @@ qlcnic_setup_netdev(struct qlcnic_adapter *adapter,
 #endif
 
 	netdev->features |= (NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_HW_CSUM |
-				NETIF_F_GRO);
+				NETIF_F_GRO | NETIF_F_VLAN_CSUM);
 
 	if (adapter->capabilities & QLCNIC_FW_CAPABILITY_TSO)
-		netdev->features |= (NETIF_F_TSO | NETIF_F_TSO6);
+		netdev->features |= (NETIF_F_TSO | NETIF_F_TSO6 |
+					NETIF_F_VLAN_TSO);
 
 	if (pci_using_dac)
 		netdev->features |= NETIF_F_HIGHDMA;
